rust-runtime/aws-smithy-http/src/lib.rs

```diff
@@ -25,7 +25,11 @@
 #![allow(clippy::derive_partial_eq_without_eq)]
 #![cfg_attr(docsrs, feature(doc_cfg))]
 
-pub mod body;
+//TODO(runtimeCratesVersioningCleanup): Re-point those who use the following reexports to
+// directly depend on `aws_smithy_types` and remove the reexports below.
+pub use aws_smithy_types::body;
+pub use aws_smithy_types::byte_stream;
```

Comment:
Should add `#[deprecated` to these with a note explaining where they moved to.

Reply:
Looks like deprecating re-exports is not supported (https://github.com/rust-lang/rust/issues/30827). Maybe changelog entries are a good alternative place?

Reply:
Type aliases work instead, e.g.
```
/// Stream of binary data
#[deprecated(note = "Moved to `aws_smithy_types::byte_stream::ByteStream`.")]
pub type ByteStream = aws_smithy_types::byte_stream::ByteStream;
```

However, adding this as part of this PR will add noise everywhere during compilation:
```
warning: use of deprecated type alias `byte_stream::ByteStream`: Moved to `aws_smithy_types::byte_stream::ByteStream`.
```

Will immediately follow-up on this PR, updating the existing places of moved types (similar to #3070).